### PR TITLE
ActiveDirectory: memberof attribute is a dn

### PR DESCRIPTION
**Pull request by sathieu**
This fixes: http://projects.theforeman.org/issues/10340

### CODE REVIEW

File: lib/ldap_fluff/ad_member_service.rb [hunk 1]
```diff
@@ -27,13 +27,11 @@ def _groups_from_ldap_data(payload)
...
     set = []
-    gids.each do |g|
-      filter = group_filter(g) & class_filter
-      search = @ldap.search(:filter => filter, :base => @group_base)
-      if !search.nil? && !search.first.nil?
-        group = search.first
+    group_dns.each do |group_dn|
+      group = @ldap.search(:base => group_dn, :scope => Net::LDAP::SearchScope_BaseObject)
+      if !group.nil?
         set  += get_groups(group[:memberof])
```
Comment (domcleal) [inline lib/ldap_fluff/ad_member_service.rb]: I think `group` is actually an array, so you might need to use `.first` or similar (see the original code which did `group = search.first`):

<pre>
2.0.0-p353 :016 > ldap.search(:base => 'ou=chemists,dc=example,dc=com', :scope => Net::LDAP::SearchScope_BaseObject)
2015-07-21T15:39:42 [ldap] [D]   op search (290.5ms)  [ filter=, base=ou=chemists,dc=example,dc=com ]
 => [#<Net::LDAP::Entry:0x0000000b1b00e8 @myhash={:dn=>["ou=chemists,dc=example,dc=com"], :ou=>["chemists"], ....]
</pre>

**Comment by domcleal** [comment]
Are you OK to update tests in test/ad_member_services_test.rb too?  It's mostly the search expectations, so changing :filter/:base to :base/:scope instead.

**Comment by sathieu** [comment]
@domcleal OK, I just need some. Will do within 3 days.

**Comment by sathieu** [comment]
I still need to fix testsuite.

**Comment by sathieu** [comment]
@domcleal I think it's OK now. _walk_group_ancestry now work on DNs only.

**Comment by domcleal** [comment]
Thanks very much @sathieu!

**Comment by swizzley** [comment]
How would one go about implementing this in a operational 1.8.2 ?

**Comment by domcleal** [comment]
Apply the patch to the gem installation, or if you're using RPMs perhaps take the latest from our nightly repositories.

**Pull request merged**